waku/node/v2/wakunode2.nim

```diff
@@ -176,30 +177,24 @@ proc unsubscribe*(w: WakuNode, contentFilter: ContentFilter) =
   ## Status: Not yet implemented.
   ## TODO Implement.
 
-proc publish*(w: WakuNode, topic: Topic, message: Message) =
-  ## Publish a `Message` to a PubSub topic.
+
+proc publish*(node: WakuNode, topic: Topic, message: WakuMessage) =
+  ## Publish a `WakuMessage` to a PubSub topic. `WakuMessage` should contain a
+  ## `contentTopic` field for light node functionality. This field may be also
+  ## be omitted.
   ##
-  ## Status: Partially implemented.
+  ## Status: Implemented.
   ##
-  ## TODO WakuMessage OR seq[byte]. NOT PubSub Message.
-  let wakuSub = w.switch.pubSub.get()
-  # XXX Consider awaiting here
-  discard wakuSub.publish(topic, message)
 
-proc publish*(w: WakuNode, topic: Topic, contentFilter: ContentFilter, message: Message) =
-  ## Publish a `Message` to a PubSub topic with a specific content filter.
-  ## Currently this means a `contentTopic`.
-  ##
-  ## Status: Not yet implemented.
-  ## TODO Implement as wrapper around `waku_relay` and `publish`.
-  ## TODO WakuMessage. Ensure content filter is in it.
+  # TODO Basic getter function for relay
+  let wakuRelay = cast[WakuRelay](node.switch.pubSub.get())
 
-  w.messages.insert((contentFilter.contentTopic, message))
+  # XXX Unclear what the purpose of this is
+  # Commenting out as it is later expected to be Message type, not WakuMessage
```

Comment:
Yeah, I guess this is only there for historical messages? So this should eventually be replaced with an optional (when historical messages is enabled) call to store the messages.

Reply:
Do we want Message or WakuMessage as part of historical messages? Not obvious to me.

cc @decanus